src/User.cpp

```diff
@@ -852,6 +852,99 @@ bool CUser::IsHostAllowed(const CString& sHostMask) const {
     for (const CString& sHost : m_ssAllowedHosts) {
         if (sHostMask.WildCmp(sHost)) {
             return true;
+        } else {
+            // CIDR notation checker, e.g. "192.0.2.0/24" or "2001:db8::/32"
+            addrinfo *aiHost, *aiRange;
+            addrinfo aiHints;
+            in6_addr in6aBitmask;
+
+            memset(&in6aBitmask, 0, sizeof(in6aBitmask));
+            memset(&aiHints, 0, sizeof(addrinfo));
+
+            aiHints.ai_family = AF_UNSPEC;  // Allow any address family
+            aiHints.ai_socktype = 0;        // Any socket
+            aiHints.ai_flags = AI_NUMERICHOST;
+            aiHints.ai_protocol = 0;  // Any protocol
+            aiHints.ai_canonname = NULL;
+            aiHints.ai_addr = NULL;
+            aiHints.ai_next = NULL;
```

Comment:
s/NULL/nullptr/

However, I don't see value in this; you already nulled it via memset. So only important fields (non-null) need to be set.